### ISSUE TITLE
Silent Failure on Fetch

### ISSUE THREAD

**Issue by xscheiner** [derbyjs/racer#213]
I'm not 100% sure this is where I should raise this, because the source of the problem is actually one of your dependencies: share. They came out with a new version last week which happened to coincide with me getting a new laptop and building a new Vagrant image on it. When I did that, this code began to fail silently:

```javascript
var isProjectNameAllowed = function (model, userId, val, cb) {
    var id = slugify(val);
    if ((val.trim() && id.replace(/-/g, '')).trim()) {
        id = userId + '::' + id;
        var project = model.at('projects.' + id.replace(/\./g, '-'));
        project.fetch(function () {
            if (!project.get()) {
                return cb(id);
            } else {
                return cb(false);
            }
        });
    } else {
        return cb(false);
    }
};
```
This method gets called twice if a user is creating a project: once while the user is typing to highlight the text field if they are typing a non-unique name, and then again when they submit to create a new project as final validation that the name really is unique before we create it.

On that second call of isProjectNameAllowed, the cb never gets called after the fetch. I have been unable to determine exactly why. Pinning share at v. 0.7.5, what was in my old dev environment, works. I have worked out that the change happened between 0.7.5 of share and 0.7.7.

I have also spent some time trying to debug this. I know for certain that in Share, if I swap the entire contents of the file lib/client/doc.js putting the 0.7.5 version in with 0.7.7, my problem goes away. I've tried to work out where the critical difference is, but I just can't. I have never been able to get node-debugger to work with derby or I could probably get a bit more detail.

I felt I should raise the defect here because the failure is really derby code that is bombing on a dependency, where share is less likely to be able to reproduce a problem where their code is being implemented in a particular way.

**Comment by nateps** [comment]
Thanks for the report. I also help to maintain ShareJS now, so I got to the bottom of this and think I fixed it in the latest version of ShareJS. Should be fixed in 0.7.12.

**Comment by xscheiner** [comment]
Thanks. I just tested and it worked beautifully.
